### PR TITLE
Pa01

### PR DESCRIPTION
**Pull request by mhmdrdwn**
Hi Daniel, we have been told that the request should be reviewed by you since me and Nasibeh are working on this project and she is second reviewer from your group.
By: Mohamed Radwan, Nasibeh Mohammadi

### CODE REVIEW

**Comment by nasibehm** [comment]
Approved

**Comment by nasibehm** [comment]
Approved

File: src/mohamed_radwan_ex/pa01/snakes_and_ladders.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    -------
+    num_moves : int
+        Number of moves the winning player needed to reach the goal
+    """
+
+    finish_point = 90
+    total_moves = 0
+    ladders = {1: 40, 8: 10, 36: 52, 43: 62, 49: 79, 65: 82, 68: 85}
+    snakes = {24: 5, 33: 3, 42: 30, 56: 37, 64: 27, 74: 12, 87: 70}
+    """positions list for all players"""
```
Comment (danielambrosius) [inline src/mohamed_radwan_ex/pa01/snakes_and_ladders.py]: This is a single line comment and should be written with `# `

File: src/mohamed_radwan_ex/pa01/snakes_and_ladders.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    position_list = [0] * num_players
+    while True:
+        for player_num in range(num_players):
+            die = random.randint(1, 6)
+            position_list[player_num] += die
+            if position_list[player_num] in ladders.keys():
```
Comment (danielambrosius) [inline src/mohamed_radwan_ex/pa01/snakes_and_ladders.py]: here you could easily concatinate the `snakes` and `ladders` dicts and use a single `if` test.

File: src/mohamed_radwan_ex/pa01/snakes_and_ladders.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                position_list[player_num] = ladders[position_list[player_num]]
+            if position_list[player_num] in snakes.keys():
+                position_list[player_num] = snakes[position_list[player_num]]
+            """total moves for all players till winner reaches the finish"""
+            total_moves += 1
+            if position_list[player_num] >= finish_point:
+                """number of rounds by subdividing number of total moves by 
+                number of players. The round up here total number of moves will 
+                always be less or equals to a factorial of number of players"""
+                num_moves = math.ceil(total_moves / num_players)
```
Comment (danielambrosius) [inline src/mohamed_radwan_ex/pa01/snakes_and_ladders.py]: This is not necessary. Just increment the moves outside the `for` loop.

**Pull request merged**